php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -1691,28 +1699,32 @@ public ReturnTypeClauseTree returnTypeClause(InternalSyntaxToken colonToken, Dec
     return new ReturnTypeClauseTreeImpl(colonToken, typeTree);
   }
 
-  public SeparatedListImpl<ExpressionTree> arguments(Optional<SeparatedListImpl<ExpressionTree>> arguments) {
+  public SeparatedListImpl<FunctionCallArgumentTree> arguments(Optional<SeparatedListImpl<FunctionCallArgumentTree>> arguments) {
     return arguments.or(SeparatedListImpl.empty());
   }
 
-  public SeparatedListImpl<ExpressionTree> argumentsList(
-    ExpressionTree firstArgument,
-    Optional<List<Tuple<InternalSyntaxToken, ExpressionTree>>> otherArguments,
+  public SeparatedListImpl<FunctionCallArgumentTree> argumentsList(
+    FunctionCallArgumentTree firstArgument,
+    Optional<List<Tuple<InternalSyntaxToken, FunctionCallArgumentTree>>> otherArguments,
     Optional<InternalSyntaxToken> trailingComma) {
     return separatedList(firstArgument, otherArguments, trailingComma.orNull());
   }
 
   public AnonymousClassTree anonymousClass(
     InternalSyntaxToken classToken,
-    Optional<InternalSyntaxToken> lParenthesis, SeparatedListImpl<ExpressionTree> arguments, Optional<InternalSyntaxToken> rParenthesis,
+    Optional<InternalSyntaxToken> lParenthesis, SeparatedListImpl<FunctionCallArgumentTree> arguments, Optional<InternalSyntaxToken> rParenthesis,
     Optional<Tuple<InternalSyntaxToken, NamespaceNameTree>> extendsClause,
     Optional<Tuple<InternalSyntaxToken, SeparatedListImpl<NamespaceNameTree>>> implementsClause,
     InternalSyntaxToken lCurlyBrace, Optional<List<ClassMemberTree>> members, InternalSyntaxToken rCurlyBrace
   ) {
+    List<ExpressionTree> argumentValues = arguments.stream()
+      .map(FunctionCallArgumentTree::value)
+      .collect(Collectors.toList());
+
     return new AnonymousClassTreeImpl(
       classToken,
       lParenthesis.orNull(),
-      arguments,
+      new SeparatedListImpl<>(argumentValues, arguments.getSeparators()),
```

Comment:
It seems that we lose the argument names to the constructor of the anonymous class, e.g.:
```
$a = class (name: $x) {
    public function __construct($name) {}
};
```
I'm not sure whether this is supported by PHP 8.

Reply:
Seems to be supported. I've adapted `AnonymousClassTree`  in this [commit](https://github.com/SonarSource/sonar-php/pull/679/commits/3b3556ceca513c8fa36e64f0aa7e2bd2aab06208). Thanks for noticing!